bank/src/main/java/com/clouway/bricky/adapter/http/BalanceService.java

```diff
@@ -0,0 +1,68 @@
+package com.clouway.bricky.adapter.http;
+
+import com.clouway.bricky.core.db.balance.BalanceRepository;
+import com.clouway.bricky.core.db.balance.FundDeficitException;
+import com.clouway.bricky.core.user.CurrentUser;
+import com.google.inject.Inject;
+import com.google.sitebricks.At;
+import com.google.sitebricks.client.transport.Json;
+import com.google.sitebricks.headless.Reply;
+import com.google.sitebricks.headless.Request;
+import com.google.sitebricks.headless.Service;
+import com.google.sitebricks.http.Post;
+
+import static javax.servlet.http.HttpServletResponse.SC_BAD_REQUEST;
+import static javax.servlet.http.HttpServletResponse.SC_CREATED;
+import static javax.servlet.http.HttpServletResponse.SC_FORBIDDEN;
+
+/**
+ * @author Marian Zlatev <mzlatev91@gmail.com>
+ */
+@Service
+@At("/r/balance")
+public class BalanceService {
+
+  private final BalanceRepository repository;
+
+  @Inject
+  public BalanceService(BalanceRepository repository) {
+    this.repository = repository;
+  }
+
+  @Post
+  @At("/deposit")
+  public Reply<?> deposit(Request request) {
+    AmountDTO dto = request.read(AmountDTO.class).as(Json.class);
+
+    if (dto.amount < 0) {
+      return Reply.with("Operation failed.").status(SC_BAD_REQUEST).as(Json.class);
+    }
```

Comment:
In REST service should not have logic. If use ```javax.validation.constraints.Pattern``` annotation we can validate amount to match to same regex (in this case pattern for numbers greater then zero). And this logic will be unnecessary.